meson.build

```diff
@@ -4,7 +4,7 @@ project(
   meson_version: '>= 1.1.0',
   default_options: [
     'buildtype=release',
-    'cpp_std=c++17',
+    'cpp_std=c++20',
```

Comment:
I didn't see any of cpp 20's date formatting, this may all just work in 17.

Reply:
It didn't compile with c++17 for me. I'm not familiar with C++ ecosystem, is the newer version problematic eg. in the python builds?

Reply:
If the system clang or gcc is new enough to handle cpp20 it shouldn't be an issue building; it's more a question of how old of a toolchain does turbodbc want to support. Since I'm not a maintainer, I'm likely not the best guy to answer that one ;-)

Reply:
C++17 on this PR gives:
```
../cpp/turbodbc/Library/turbodbc/field.h:14:28: error: 'year_month_day' in namespace 'std::chrono' does not name a type
   14 | using date  = std::chrono::year_month_day;```

Reply:
Turbodbc mostly follows the SPEC-0 and NEP-29 which means python 3.11 support.
The OS versions shipping with python 3.11 also have gcc 11 which supports C++20 chrono. 
The OS versions in the CI matrix also passed, so this should be good (also it's not something hard to revert partially or completely).